### PR TITLE
Made a few enhancements, some are a bit rough

### PR DESCRIPTION
**Pull request by johnkeates**
Added: 
- Jinja templated configuration
- enable haproxy after installation on debian and ubuntu
- added a few lines for documentation

This leaves for a few to-be-desired features:
- More or all configuration options via a pillar
- a map file to make sure it works with more distros
- either over-engineer all configuration options or none (currently, it's not completely levelled out)

### CODE REVIEW

**Comment by nmadhok** [comment]
@johnkeates the standard for salt-formulas README file is to use Restructured Text Syntax instead of Markdown syntax. Can you change it to rst?

**Comment by johnkeates** [comment]
Sure, I'll revert it. I didn't know about the standards, and the formatting went all awry with RST to I made an MD out of it. Is there some sort of a reference guide on committing to formulas that are in this repo?

**Comment by gravyboat** [comment]
@johnkeates There is http://docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#writing-formulas and docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#repository-structure .

**Comment by johnkeates** [comment]
Changed the documentation formatting, should be good now. Didn't have the time to completely read the docs, will do so for future formula updates.

File: README.rst [hunk 1]
```diff
@@ -1,7 +1,57 @@
 haproxy
 =======
 
-haproxy
--------
+**haproxy:**
```
Comment (nmadhok) [inline README.rst]: You forgot to change this back to what it was earlier.

**Comment by nmadhok** [comment]
Thanks @gravyboat for link to the docs. @johnkeates Can you change what I've pointed out in the line note?

File: README.rst (hunk 1 above)
Comment (johnkeates) [inline README.rst]: Ah yes, I see. Changed it back.

**Comment by johnkeates** [comment]
I also made a small fix with the bind statement because on HAProxy 1.5 with SSL support it randomly starts spitting out HTTP over 443.

**Pull request merged**